substratest/factory.py

```diff
@@ -479,12 +498,14 @@ def create_dataset(self, objective=None, permissions=None):
             name=name,
             data_opener=str(opener_path),
             type='Test',
+            metadata=metadata or {},
             description=str(description_path),
             objective_key=objective.key if objective else None,
             permissions=permissions or DEFAULT_PERMISSIONS,
         )
 
-    def create_objective(self, dataset=None, data_samples=None, permissions=None):
+    def create_objective(self, dataset=None, data_samples=None, permissions=None, metadata=None):
+        metadata = metadata or {}
```

Comment:
The pattern you're using here is different from the one you used for compute plan testtuples. You should chose one and apply it throughout the file.